Coursemology/coursemology2
# Add Youtube Captions

Add a button to toggle captions on Youtube on and off.

Youtube doesn't actually have a proper API for this, so we do this by forcing the captions on in the Youtube options, and load and unload the captions based on a Redux state: https://stackoverflow.com/questions/13735783/youtube-api-how-to-use-custom-controls-to-turn-captions-on-off-change-languag

There's also a possibility that Youtube doesn't have captions to show, in which case the `captions` option will be missing. We have a initial "NOT_LOADED" state where the captions button isn't shown. If no captions are found, then the state will always remain as "NOT_LOADED" and the button will never show. If captions are found after video starts, then the button shows. This mirrors Youtube behaviour (somewhat).

Test plan:
![image](https://user-images.githubusercontent.com/3049761/91812560-28ef5380-ebe6-11ea-9f98-6d2e7b01e80a.png)
Check the cc button is there, bolded when captions are on and showing, grey when off.

There's a few scenarios we need to go through:
> For context, without explicitly forcing captions, embedded YouTube actually remembers your caption settings on the main site.

1) Check that no matter what is set on youtube.com for captions, the player follows the state of the cc button to show or hide captions
2) Find a video with no captions (on youtube.com the captions button won't show either). Check that the caption button doesn't show either
3) Check that the cc toggle state on coursemology doesn't affect youtube.com and vice-versa
4) Play/pause/seek and make sure everything still works.
